lib/mvi/messages/message_builder.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+require 'ox'
+
+module MVI
+  module Messages
+    module MessageBuilder
+      def doc
+        @doc ||= Ox::Document.new(version: '1.0')
+      end
+
+      def header(extension)
+        @message = idm(extension)
```

Comment:
good point, I had this originally as an included module, so the message and doc were instance vars probably safer to change it back.